### ISSUE TITLE
Thinq1 Mini-Split AC: TypeError: Cannot read property 'min' of null

### ISSUE THREAD

**Issue by coneybeare** [nVuln/homebridge-lg-thinq#66]
**Describe The Bug:**
When enabling Thinq1 support, used against a LSN120HSV5 mini-split unit. It uses Thinq1

**To Reproduce:**
Enabling the plugin raises an error when trying to configure the device:

**Expected behavior:**

I was hoping to get control over this Thinq1 device

**Logs:**

```
9/2/2021, 7:09:53 AMLg Thinq Bridge starting
9/2/2021, 7:09:57 AMLg Thinq BridgeLoaded plugin 'homebridge-lg-thinq'
9/2/2021, 7:09:57 AMLg Thinq BridgeLoading 1 platforms...
9/2/2021, 7:09:57 AMLg Thinq BridgeLGThinQLoading accessory from cache: Master Bedroom
9/2/2021, 7:09:57 AMLg Thinq BridgeLGThinQLoading accessory from cache: Living Room
9/2/2021, 7:09:57 AMLg Thinq BridgeBridge is running on port 51846.
9/2/2021, 7:10:04 AMLg Thinq BridgeLGThinQRestoring existing accessory from cache: Master Bedroom
9/2/2021, 7:10:04 AMLg Thinq BridgeLGThinQAC still in development.If you got problem,plz report at https://github.com/nVuln/homebridge-lg-thinq/issues
9/2/2021, 7:10:05 AMLg Thinq BridgeLGThinQAdding new accessory: Dining Room
9/2/2021, 7:10:05 AMLg Thinq BridgeLGThinQAC still in development.If you got problem,plz report at https://github.com/nVuln/homebridge-lg-thinq/issues
9/2/2021, 7:10:05 AMLg Thinq BridgeWARNINGunhandled rejection: TypeError: Cannot read property 'min' of null
```

**Plugin Config:**

```json
{
    "accessories": [],
    "platforms": [
        {
            "platform": "LGThinQ",
            "country": "US",
            "language": "en-US",
            "refresh_interval": 60,
            "auth_mode": "account",
            "username": "REDACTED",
            "password": "REDACTED",
            "thinq1": true,
            "refresh_token": ""
        }
    ]
}
```

**Screenshots:**

**Environment:**

* **Plugin Version**: 1.1.2
* **Homebridge Version**: 1.3.4 via hoobs 4.0.100
* **Node.js Version**: 14.17.5
* **NPM Version**: 6.14.14
* **Operating System**: Raspbian via hoobs

```
$ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

**Comment by nVuln** [comment]
turn on homebridge debug mode, you can see some debug logging, please upload all

**Comment by coneybeare** [comment]
```
9/2/2021, 7:51:29 AMLg Thinq BridgeLGThinQFound device:  {"deviceId":"REDACTED-4d23-1f8f-aa61-60ab14f44252","deviceType":401,"modelName":"RAC_056905_WW","subModelNm":null,"sensorType":null,"alias":"Master Bedroom","deviceCode":"AI01","networkType":"02","tftYn":"N","guideTypeYn":"Y","guideType":"RAC_TYPE1","pccModelYn":"N","autoOrderYn":"N","drServiceYn":"N","ssid":"REDACTED","timezoneCode":"America/New_York","timezoneCodeAlias":"USA/New_York","sdsGuide":"{\"deviceCode\":\"AI01\"}","newRegYn":"N","remoteControlType":"","fareTarget":null,"area":"334505","sleep":null,"deviceState":"E","rmsClientId":null,"regDtUtc":"20201208142938000","regIndex":0,"blackboxYn":"Y","groupableYn":"Y","controllableYn":"Y","combinedProductYn":"N","masterYn":"Y","snapshot":{"airState.windStrength":6,"airState.wMode.lowHeating":0,"airState.diagCode":0,"airState.lightingState.displayControl":1,"airState.wDir.hStep":0,"mid":862065947,"airState.energy.onCurrent":0,"airState.wMode.airClean":0,"airState.quality.sensorMon":0,"airState.miscFuncState.antiBugs":0,"airState.tempState.target":26,"airState.operation":1,"airState.wMode.jet":0,"airState.wDir.vStep":100,"timestamp":1630581272922,"airState.powerSave.basic":0,"airState.quality.PM10":0,"static":{"deviceType":"401","countryCode":"US"},"airState.quality.overall":0,"airState.tempState.current":26,"airState.miscFuncState.extraOp":0,"airState.reservation.sleepTime":0,"airState.miscFuncState.autoDry":0,"airState.reservation.targetTimeToStart":0,"meta":{"allDeviceInfoUpdate":false,"messageId":"gViT80JySWWmOonhB7ZpKg"},"airState.quality.PM1":0,"airState.quality.PM2":0,"online":true,"airState.opMode":0,"airState.reservation.targetTimeToStop":0,"airState.filterMngStates.maxTime":360,"airState.filterMngStates.useTime":220},"manufacture":null,"online":true,"platformType":"thinq2","homeDeviceOrder":-1,"roomDeviceOrder":-1,"ownershipYn":"Y","modelJsonVer":"11.74","modelJsonUri":"https://objectcontent.lgthinq.com/REDACTED-a6cb-4ad8-8028-13cffdb83d54?hdnts=exp=1689007282~hmac=95472632ce69 REDACTED 8286d0a6aed8b97054a29d137015525df","appModuleVer":"16.32","appModuleUri":"https://objectcontent.lgthinq.com/REDACTED-0404-4712-a5bc-2bc3d9e072b8?hdnts=exp=1685169055~hmac=8b53 REDACTED 3d075c761cc94be01589d82200ab3e93c3fa57","appRestartYn":"Y","appModuleSize":"13305760","langPackProductTypeVer":"52.8","langPackProductTypeUri":"https://objectcontent.lgthinq.com/REDACTED-1ce6-4ded-ac44-2d450cea341a?hdnts=exp=1693362382~hmac=e74e REDACTED 8d2915198852aa8b2bc8a5797a9a8cb92d8fc12e2ff6212","langPackModelVer":null,"langPackModelUri":null,"roomId":"","fwInfoList":[{"checksum":"00006742","partNumber":"SAA38690428","order":1}],"modemInfo":{"modelName":"RAC_056905_WW","appVersion":"clip_hna_v1.9.104","modemType":"QCOM_QCA4010","ruleEngine":"y"},"existsEntryPopup":"N","fwVer":null,"modemVer":null,"subDeviceCount":0,"firebaseLogKey":"P:SD","cardType":"Small","cardControl":"Base"}
9/2/2021, 7:51:29 AMLg Thinq BridgeLGThinQRestoring existing accessory from cache: Master Bedroom
9/2/2021, 7:51:29 AMLg Thinq BridgeLGThinQAC still in development.If you got problem,plz report at https://github.com/nVuln/homebridge-lg-thinq/issues
9/2/2021, 7:51:29 AMLg Thinq BridgeLGThinQFound device:  {"deviceId":"REDACTED-7149-11d3-80b2-7440be93575f","deviceType":401,"modelName":"RAC_056905_WW","subModelNm":null,"sensorType":null,"alias":"Dining Room","deviceCode":"AI01","networkType":"02","tftYn":"N","guideTypeYn":"Y","guideType":"RAC_TYPE1","pccModelYn":"N","autoOrderYn":"N","drServiceYn":"N","ssid":"REDACTED","timezoneCode":"America/New_York","timezoneCodeAlias":"USA/New_York","sdsGuide":"{\"deviceCode\":\"AI01\"}","newRegYn":"N","remoteControlType":"","fareTarget":null,"area":"334505","sleep":null,"deviceState":"E","rmsClientId":null,"regDtUtc":"20201208141858000","regIndex":0,"blackboxYn":null,"groupableYn":"Y","controllableYn":"Y","combinedProductYn":"N","masterYn":"Y","manufacture":null,"online":false,"platformType":"thinq1","homeDeviceOrder":-1,"roomDeviceOrder":-1,"ownershipYn":"Y","modelJsonVer":"7.12","modelJsonUri":"https://aic.lgthinq.com:46030/api/webContents/modelJSON?modelName=modelJSON_401&countryCode=KR&contentsId= REDACTED-7149-11d3-80b2-7440be93575f&authKey=thinq","appModuleVer":"16.32","appModuleUri":"https://objectcontent.lgthinq.com/REDACTED-0404-4712-a5bc-2bc3d9e072b8?hdnts=exp=1685169055~hmac=8b538678 REDACTED 61cc94be01589d82200ab3e93c3fa57","appRestartYn":"Y","appModuleSize":"13305760","langPackProductTypeVer":"52.8","langPackProductTypeUri":"https://objectcontent.lgthinq.com/REDACTED-1ce6-4ded-ac44-2d450cea341a?hdnts=exp=1693362382~hmac=e74ed08 REDACTED 198852aa8b2bc8a5797a9a8cb92d8fc12e2ff6212","langPackModelVer":"","langPackModelUri":"","roomId":"","fwInfoList":null,"modemInfo":null,"existsEntryPopup":"N","fwVer":"2.5.8_RTOS_3K","modemVer":null,"subDeviceCount":0,"firebaseLogKey":"P:SD","cardType":"Small","cardControl":"Base"}
9/2/2021, 7:51:29 AMLg Thinq BridgeLGThinQAdding new accessory: Dining Room
9/2/2021, 7:51:29 AMLg Thinq BridgeLGThinQAC still in development.If you got problem,plz report at https://github.com/nVuln/homebridge-lg-thinq/issues
9/2/2021, 7:51:29 AMLg Thinq BridgeWARNINGunhandled rejection: TypeError: Cannot read property 'min' of null
9/2/2021, 9:10:00 AMlgthinqbridge stopping
9/2/2021, 9:10:00 AMLg Thinq BridgeShutting down
9/2/2021, 9:10:02 AMLg Thinq BridgeStopped
9/2/2021, 9:10:05 AMLg Thinq Bridge starting
9/2/2021, 9:10:08 AMLg Thinq BridgeInitializing HAP-NodeJS v0.9.4...
9/2/2021, 9:10:10 AMLg Thinq BridgeLoaded plugin 'homebridge-lg-thinq'
9/2/2021, 9:10:10 AMLg Thinq BridgeRegistering platform 'homebridge-lg-thinq.LGThinQ'
9/2/2021, 9:10:10 AMLg Thinq BridgeLoading 1 platforms...
9/2/2021, 9:10:10 AMLg Thinq BridgeLGThinQLoading accessory from cache: Master Bedroom
9/2/2021, 9:10:10 AMLg Thinq BridgeLGThinQLoading accessory from cache: Living Room
9/2/2021, 9:10:10 AMLg Thinq BridgeLGThinQExecuted didFinishLaunching callback
9/2/2021, 9:10:10 AMLg Thinq BridgePreparing Advertiser for 'Lg Thinq Bridge 76F6' using bonjour-hap backend!
9/2/2021, 9:10:10 AMLg Thinq BridgeSetup URI 'X-HM://0023ISYWY7B46'
9/2/2021, 9:10:10 AMLg Thinq BridgeStarting to advertise 'Lg Thinq Bridge 76F6' using bonjour-hap backend!
9/2/2021, 9:10:10 AMLg Thinq BridgeBridge is running on port 51846.
9/2/2021, 9:10:17 AMLg Thinq BridgeLGThinQFound device:  {"deviceId":"REDACTED-4d23-1f8f-aa61-60ab14f44252","deviceType":401,"modelName":"RAC_056905_WW","subModelNm":null,"sensorType":null,"alias":"Master Bedroom","deviceCode":"AI01","networkType":"02","tftYn":"N","guideTypeYn":"Y","guideType":"RAC_TYPE1","pccModelYn":"N","autoOrderYn":"N","drServiceYn":"N","ssid":"REDACTED","timezoneCode":"America/New_York","timezoneCodeAlias":"USA/New_York","sdsGuide":"{\"deviceCode\":\"AI01\"}","newRegYn":"N","remoteControlType":"","fareTarget":null,"area":"334505","sleep":null,"deviceState":"E","rmsClientId":null,"regDtUtc":"20201208142938000","regIndex":0,"blackboxYn":"Y","groupableYn":"Y","controllableYn":"Y","combinedProductYn":"N","masterYn":"Y","snapshot":{"airState.windStrength":6,"airState.wMode.lowHeating":0,"airState.diagCode":0,"airState.lightingState.displayControl":1,"airState.wDir.hStep":0,"mid":865718922,"airState.energy.onCurrent":0,"airState.wMode.airClean":0,"airState.quality.sensorMon":0,"airState.miscFuncState.antiBugs":0,"airState.tempState.target":26,"airState.operation":1,"airState.wMode.jet":0,"airState.wDir.vStep":100,"timestamp":1630584840435,"airState.powerSave.basic":0,"airState.quality.PM10":0,"static":{"deviceType":"401","countryCode":"US"},"airState.quality.overall":0,"airState.tempState.current":26,"airState.miscFuncState.extraOp":0,"airState.reservation.sleepTime":0,"airState.miscFuncState.autoDry":0,"airState.reservation.targetTimeToStart":0,"meta":{"allDeviceInfoUpdate":false,"messageId":"Qw4lpenXSqOK7QdHYBadTw"},"airState.quality.PM1":0,"airState.quality.PM2":0,"online":true,"airState.opMode":0,"airState.reservation.targetTimeToStop":0,"airState.filterMngStates.maxTime":360,"airState.filterMngStates.useTime":219},"manufacture":null,"online":true,"platformType":"thinq2","homeDeviceOrder":-1,"roomDeviceOrder":-1,"ownershipYn":"Y","modelJsonVer":"11.74","modelJsonUri":"https://objectcontent.lgthinq.com/REDACTED-a6cb-4ad8-8028-13cffdb83d54?hdnts=exp=1689007282~hmac=95472632ce REDACTED 1b368286d0a6aed8b97054a29d137015525df","appModuleVer":"16.32","appModuleUri":"https://objectcontent.lgthinq.com/REDACTED-0404-4712-a5bc-2bc3d9e072b8?hdnts=exp=1685169055~hmac=8b53867 REDACTED c761cc94be01589d82200ab3e93c3fa57","appRestartYn":"Y","appModuleSize":"13305760","langPackProductTypeVer":"52.8","langPackProductTypeUri":"https://objectcontent.lgthinq.com/REDACTED-1ce6-4ded-ac44-2d450cea341a?hdnts=exp=1693362382~hmac=e74ed0863 REDACTED 198852aa8b2bc8a5797a9a8cb92d8fc12e2ff6212","langPackModelVer":null,"langPackModelUri":null,"roomId":"","fwInfoList":[{"checksum":"00006742","partNumber":"SAA38690428","order":1}],"modemInfo":{"modelName":"RAC_056905_WW","appVersion":"clip_hna_v1.9.104","modemType":"QCOM_QCA4010","ruleEngine":"y"},"existsEntryPopup":"N","fwVer":null,"modemVer":null,"subDeviceCount":0,"firebaseLogKey":"P:SD","cardType":"Small","cardControl":"Base"}
9/2/2021, 9:10:17 AMLg Thinq BridgeLGThinQRestoring existing accessory from cache: Master Bedroom
9/2/2021, 9:10:17 AMLg Thinq BridgeLGThinQAC still in development.If you got problem,plz report at https://github.com/nVuln/homebridge-lg-thinq/issues
9/2/2021, 9:10:17 AMLg Thinq BridgeLGThinQFound device:  {"deviceId":"REDACTED-7149-11d3-80b2-7440be93575f","deviceType":401,"modelName":"RAC_056905_WW","subModelNm":null,"sensorType":null,"alias":"Dining Room","deviceCode":"AI01","networkType":"02","tftYn":"N","guideTypeYn":"Y","guideType":"RAC_TYPE1","pccModelYn":"N","autoOrderYn":"N","drServiceYn":"N","ssid":"REDACTED-IOT","timezoneCode":"America/New_York","timezoneCodeAlias":"USA/New_York","sdsGuide":"{\"deviceCode\":\"AI01\"}","newRegYn":"N","remoteControlType":"","fareTarget":null,"area":"334505","sleep":null,"deviceState":"E","rmsClientId":null,"regDtUtc":"20201208141858000","regIndex":0,"blackboxYn":null,"groupableYn":"Y","controllableYn":"Y","combinedProductYn":"N","masterYn":"Y","manufacture":null,"online":false,"platformType":"thinq1","homeDeviceOrder":-1,"roomDeviceOrder":-1,"ownershipYn":"Y","modelJsonVer":"7.12","modelJsonUri":"https://aic.lgthinq.com:46030/api/webContents/modelJSON?modelName=modelJSON_401&countryCode=KR&contentsId= REDACTED-7149-11d3-80b2-7440be93575f&authKey=thinq","appModuleVer":"16.32","appModuleUri":"https://objectcontent.lgthinq.com/6c1a78d3-0404-4712-a5bc-2bc3d9e072b8?hdnts=exp=1685169055~hmac=8b538 REDACTED 929b66403d075c761cc94be01589d82200ab3e93c3fa57","appRestartYn":"Y","appModuleSize":"13305760","langPackProductTypeVer":"52.8","langPackProductTypeUri":"https://objectcontent.lgthinq.com/5a0c1923-1ce6-4ded-ac44-2d450cea341a?hdnts=exp=1693362382~hmac=e74ed0863a REDACTED 2915198852aa8b2bc8a5797a9a8cb92d8fc12e2ff6212","langPackModelVer":"","langPackModelUri":"","roomId":"","fwInfoList":null,"modemInfo":null,"existsEntryPopup":"N","fwVer":"2.5.8_RTOS_3K","modemVer":null,"subDeviceCount":0,"firebaseLogKey":"P:SD","cardType":"Small","cardControl":"Base"}
9/2/2021, 9:10:17 AMLg Thinq BridgeLGThinQAdding new accessory: Dining Room
9/2/2021, 9:10:17 AMLg Thinq BridgeLGThinQAC still in development.If you got problem,plz report at https://github.com/nVuln/homebridge-lg-thinq/issues
9/2/2021, 9:10:17 AMLg Thinq BridgeWARNINGunhandled rejection: TypeError: Cannot read property 'min' of null
```

**Comment by nVuln** [comment]
> 9/2/2021, 9:10:17 AMLg Thinq BridgeLGThinQFound device:  {"deviceId":"REDACTED-7149-11d3-80b2-7440be93575f","deviceType":401,"modelName":"RAC_056905_WW","subModelNm":null,"sensorType":null,"alias":"Dining Room","deviceCode":"AI01","networkType":"02","tftYn":"N","guideTypeYn":"Y","guideType":"RAC_TYPE1","pccModelYn":"N","autoOrderYn":"N","drServiceYn":"N","ssid":"REDACTED-IOT","timezoneCode":"America/New_York","timezoneCodeAlias":"USA/New_York","sdsGuide":"{\"deviceCode\":\"AI01\"}","newRegYn":"N","remoteControlType":"","fareTarget":null,"area":"334505","sleep":null,"deviceState":"E","rmsClientId":null,"regDtUtc":"20201208141858000","regIndex":0,"blackboxYn":null,"groupableYn":"Y","controllableYn":"Y","combinedProductYn":"N","masterYn":"Y","manufacture":null,"online":false,"platformType":"thinq1","homeDeviceOrder":-1,"roomDeviceOrder":-1,"ownershipYn":"Y","modelJsonVer":"7.12","modelJsonUri":"https://aic.lgthinq.com:46030/api/webContents/modelJSON?modelName=modelJSON_401&countryCode=KR&contentsId= REDACTED-7149-11d3-80b2-7440be93575f&authKey=thinq","appModuleVer":"16.32","appModuleUri":"https://objectcontent.lgthinq.com/6c1a78d3-0404-4712-a5bc-2bc3d9e072b8?hdnts=exp=1685169055~hmac=8b538 REDACTED 929b66403d075c761cc94be01589d82200ab3e93c3fa57","appRestartYn":"Y","appModuleSize":"13305760","langPackProductTypeVer":"52.8","langPackProductTypeUri":"https://objectcontent.lgthinq.com/5a0c1923-1ce6-4ded-ac44-2d450cea341a?hdnts=exp=1693362382~hmac=e74ed0863a REDACTED 2915198852aa8b2bc8a5797a9a8cb92d8fc12e2ff6212","langPackModelVer":"","langPackModelUri":"","roomId":"","fwInfoList":null,"modemInfo":null,"existsEntryPopup":"N","fwVer":"2.5.8_RTOS_3K","modemVer":null,"subDeviceCount":0,"firebaseLogKey":"P:SD","cardType":"Small","cardControl":"Base"}

can you help me download `modelJsonUri` in this device data, because you redacted it so I can't find a correct url to download it
didn't know why 2 your device is same model `RAC_056905_WW`, but different data

**Comment by coneybeare** [comment]
Yeah sorry about the over redaction, I thought the UUID would have personal info in it. I have 3 units, 2 of them the same, one of them different.

https://aic.lgthinq.com:46030/api/webContents/modelJSON?modelName=modelJSON_401&countryCode=KR&contentsId=d27c2920-7149-11d3-80b2-7440be93575f&authKey=thinq
https://aic.lgthinq.com:46030/api/webContents/modelJSON?modelName=modelJSON_401&countryCode=KR&contentsId=d27c2920-7149-11d3-80b2-7440be93575f&authKey=thinq

After closer inspection of the alias names, It appears that I might have mislabeled the Dining Room one here. And the 2 that are here are showing up in home bridge (not accepting commands, but that is another issue I will open separately). I don't see json output for the one not working. The one that is not here is an LSN180HSV5 model, using thinq1. It is working in the LG app and I did have it working with the LG Airco plugin, so I know it is possible.

I am a software engineer too and am happy to help debug this issue with you if you have the time.

**Comment by nVuln** [comment]
I already published test version `v1.2.0-test.8`, please install and try

**Comment by coneybeare** [comment]
Genius. works well with this version. Thank you so much for taking this on! Where is your `donate` button?!
